samples/bernoulli.cr

```diff
@@ -0,0 +1,63 @@
+# Compute the Bernoulli numbers, the worlds first computer program
+# Taken from the 'Ada 99' project, https://marquisdegeek.com/code_ada99
+
+class Fraction
+  def initialize(n : Int64, d : Int64)
+    @numerator = n
+    @denominator = d
+  end
+
+  def numerator
+    @numerator
+  end
+
+  def denominator
+    @denominator
+  end
+
+  def subtract(rhs_fraction)
+    rhs_numerator = rhs_fraction.numerator * @denominator
+    rhs_denominator = rhs_fraction.denominator * @denominator
+    @numerator *= rhs_fraction.denominator
+    @denominator *= rhs_fraction.denominator
+    @numerator -= rhs_numerator
+    self.reduce
+  end
+
+  def multiply(value)
+    @numerator *= value
+  end
+
+  def reduce
+    gcd = gcd(@numerator, @denominator)
```

Comment:
I would keep using the alternative gcd impl for examples sake. Make a note that Int#gcd